src/components/application_manager/test/rpc_service_impl_test.cc

```diff
@@ -301,12 +331,362 @@ TEST_F(RPCServiceImplTest,
+  auto function_id = mobile_apis::FunctionID::OnHMIStatusID;
+  auto message_type = mobile_apis::messageType::notification;
+  MobileAPIDerived derived_api;
+  derived_api.SetSchema(function_id, message_type);
+  SetMobileAPI(derived_api);
+  CSmartSchema schema;
+  mobile_so_factory_.GetSchema(function_id, message_type, schema);
+  auto initial_schema_item = schema.getSchemaItem();
+  // Change state of schema and add custom parameter
+  auto new_function_id = mobile_apis::FunctionID::OnButtonEventID;
+  std::map<std::string, SMember> members;
+  members["custom member"] = SMember();
+  rpc_service_->UpdateMobileRPCParams(new_function_id, message_type, members);
+  mobile_so_factory_.GetSchema(new_function_id, message_type, schema);
+  // Check that schema is changed
+
+  EXPECT_NE(initial_schema_item, schema.getSchemaItem());
+}
+
+TEST_F(RPCServiceImplTest,
+       UpdateHMIRPCParams_FunctionSchemaIsResettedAndAddedCustomMember) {
+  using namespace ns_smart_device_link::ns_smart_objects;
+  // Get initial state of schema
+  auto function_id = hmi_apis::FunctionID::Buttons_OnButtonEvent;
+  auto message_type = hmi_apis::messageType::notification;
+  HMIAPIDerived derived_api;
+  derived_api.SetSchema(function_id, message_type);
+  SetHMIAPI(derived_api);
+  CSmartSchema schema;
+  hmi_so_factory_.GetSchema(function_id, message_type, schema);
+  auto initial_schema_item = schema.getSchemaItem();
+  // Change state of schema and add custom parameter
+  auto new_function_id = hmi_apis::FunctionID::Buttons_OnButtonPress;
+  std::map<std::string, SMember> members;
+  members["custom member"] = SMember();
+  rpc_service_->UpdateHMIRPCParams(function_id, message_type, members);
+  hmi_so_factory_.GetSchema(new_function_id, message_type, schema);
+  // Check that schema is changed
+
+  EXPECT_NE(initial_schema_item, schema.getSchemaItem());
+}
+
+TEST_F(RPCServiceImplTest, SendMessageToHMI_InvalidMessageType) {
+  auto message = CreateMessage(smart_objects::SmartType_Invalid);
+  (*message)[am::strings::msg_params][am::strings::function_id] = kFunctionId;
+  // hmi_apis::FunctionID::BasicCommunication_OnSystemCapabilityUpdated;
+
+  utils::Optional<am::plugin_manager::RPCPlugin> mock_rpc_plugin_opt(
+      mock_rpc_plugin_);
+  EXPECT_CALL(mock_app_mngr_, GetPluginManager())
+      .WillOnce(ReturnRef(mock_rpc_plugin_manager_));
+  EXPECT_CALL(mock_rpc_plugin_manager_, FindPluginToProcess(_, _))
+      .WillOnce(Return(mock_rpc_plugin_opt));
+  EXPECT_CALL(mock_rpc_plugin_, PluginName()).WillOnce(Return(""));
+  rpc_service_->SendMessageToHMI(message);
+}
+
+TEST_F(RPCServiceImplTest, SendMessageToHMI_NoHMIHandler) {
+  auto message = CreateMessage(smart_objects::SmartType_Invalid);
+  PrepareBasicMessage(message);
+  (*message)[am::strings::msg_params][am::strings::result_code] =
+      ns_smart_device_link_rpc::V1::Result::SUCCESS;
+  rpc_service_->set_hmi_message_handler(nullptr);
+  EXPECT_CALL(mock_app_mngr_, GetPluginManager()).Times(0);
```

Comment:
I do not see call of `SendMessageToHMI`. If hmi_message_handler was stetted to null, you should wait that hmi_message_handler will no be touched during SendMessageToHMI execution

Reply:
Yes, you are right, I added this call in commit b66aa723e29b0d16f37bb6ccaa764e5a717324a0